fakeserver/fakeserver_test.go

```diff
@@ -0,0 +1,152 @@
+package fakeserver
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/Betterment/testtrack-cli/fakeassignments"
+
+	"encoding/json"
+
+	"github.com/stretchr/testify/require"
+)
+
+var testSchema = `
+serializer_version: 1
+schema_version: "2020011774023"
+splits:
+- name: test.test_experiment
+  weights:
+    control: 60
+    treatment: 40
+`
+
+func TestMain(m *testing.M) {
+	current, exists := os.LookupEnv("TESTTRACK_FAKE_SERVER_CONFIG_DIR")
+
+	dir, err := ioutil.TempDir("", "testtrack-cli")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	schemasDir := filepath.Join(dir, "schemas")
+	if err := os.MkdirAll(schemasDir, 0755); err != nil {
+		log.Fatal(err)
+	}
+
+	schemaContent := []byte(testSchema)
+	if err := ioutil.WriteFile(filepath.Join(schemasDir, "test.yml"), schemaContent, 0644); err != nil {
+		log.Fatal(err)
+	}
+
+	os.Setenv("TESTTRACK_FAKE_SERVER_CONFIG_DIR", dir)
+	exitCode := m.Run()
+	if exists {
+		os.Setenv("TESTTRACK_FAKE_SERVER_CONFIG_DIR", current)
```

Comment:
So the only way that this wouldn't happen is if the entire process panics and bails out before it can clean up -- something I imagine that is less likely in go than it might be in a language where we would have to lean on exceptions for control flow, etc?

Reply:
yeah, something like that, but really i was banking on the test framework providing some safety here. it shouldn't start the next test if it can't tear down the current one.